### PR TITLE
wallust: 2.10.0 -> 3.0.0-beta

### PR DESCRIPTION
**Pull request by mewoocat**
## Description of changes

Change log: https://codeberg.org/explosion-mental/wallust/releases/tag/3.0.0-beta

Closes #288779

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by iynaix** [comment]
Result of `nixpkgs-review pr 318687` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wallust</li>
  </ul>
</details>

**Comment by iynaix** [comment]
Welcome to nixpkgs! :tada: 

Please change both the title of the PR as well as the commit message to `wallust: 2.10.0 -> 3.0.0-beta` as per the [commit conventions](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions)

**Comment by JohnRTitor** [comment]
Hello @mewoocat, thanks for your contribution! Could you update the commit message using `git commit --amend` and then force push the changes?

**Comment by JohnRTitor** [comment]
@ofborg eval

**Comment by mewoocat** [comment]
> Hello @mewoocat, thanks for your contribution! Could you update the commit message using `git commit --amend` and then force push the changes?

Done

**Review by iynaix — APPROVED** [review]

**Comment by iynaix** [comment]
LGTM

File: pkgs/applications/misc/wallust/default.nix [hunk 1]
```diff
@@ -17,10 +17,10 @@ rustPlatform.buildRustPackage {
...
     repo = "wallust";
     rev = version;
-    hash = "sha256-0kPmr7/2uVncpCGVOeIkYlm2M0n9+ypVl7bQ9HnqLb4=";
+    hash = "sha256-gGyxRdv2I/3TQWrTbUjlJGsaRv4SaNE+4Zo9LMWmxk8";
   };
 
-  cargoHash = "sha256-p1NKEppBYLdCsTY7FHPzaGladLv5HqIVNJxSoFJOx50=";
+  cargoHash = "sha256-dkHS8EOzmn5VLiKP3SMT0ZGAsk2wzvQeioG7NuGGUzA=";
 
   nativeBuildInputs = [ makeWrapper ];
```
Comment (JohnRTitor) [inline pkgs/applications/misc/wallust/default.nix:25]:
```suggestion
  nativeBuildInputs = [ makeWrapper installShellFiles ];

  postInstall = ''
    installManPage man/wallust*
    installShellCompletion --cmd wallust \
      --bash completions/wallust.bash \
      --zsh completions/_wallust \
      --fish completions/wallust.fish
  '';
```

File: pkgs/applications/misc/wallust/default.nix [hunk 2]
```diff
@@ -6,7 +6,7 @@
 , makeWrapper
```
Comment (JohnRTitor) [inline pkgs/applications/misc/wallust/default.nix:6]:
```suggestion
, makeWrapper
, installShellFiles
```

**Review by JohnRTitor — COMMENTED** [review]
Please also add shell completion like in #288779.

**Comment by JohnRTitor** [comment]
@ofborg build wallust

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 318687` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wallust</li>
  </ul>
</details>

**Review by JohnRTitor — APPROVED** [review]

**Review by iynaix — APPROVED** [review]

**Pull request merged**